File: 240717/흰검 칠하기/painting-white-black.py
n = int(input())

visited = [0] *(200)
color = ['Yellow'] *(200)
now = 100 
for _ in range(n):
    s,r = input().split() 
    s = int(s)
    
    if r =='L':
        for i in range(now,now-s,-1):
            visited[i]+=1
            color[i] = 'White'
        now = now - s +1
    elif r == 'R':
        for i in range(now,now+s):
            visited[i]+=1
            color[i] = 'Black'
        now=now+s-1

for i in range(200):
    if visited[i] >=4:
        color[i] ='gray'

white = color.count('White')
gray = color.count('gray')
Black = color.count("Black")
print(white,Black,gray)

File: 241202/흰검 칠하기/painting-white-black.java
import java.util.*;
public class Main {
    private static String[] color = new String[1000001];
    private static int[] blackcolor = new int[1000001];
    private static int[] yellowcolor = new int[1000001];
    private static int[] first = new int[1000001];
    private static int white,black,gray,now;
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();

        now = 100000;
        for(int s = 0; s<n; s++){
            int number = sc.nextInt();
            char word = sc.next().charAt(0);
            if(word =='R'){
                for(int i = now; i<now+number; i++ ){
                    blackcolor[i] = blackcolor[i]+1;
                    color[i] ="black";

                }
                now = now + number-1;
            }else{
                for(int i = now; i>now-number; i--){
                    yellowcolor[i] = yellowcolor[i]+1;
                    color[i] ="white";
                }
                now = now - number+1;
            }
        }

        for(int i = 0; i<1000001; i++){
            if(blackcolor[i]>=2 && yellowcolor[i]>=2){
                gray++;
            }else if(color[i]=="black"){
                black++;
            }else if(color[i]=="white"){
                white++;
            }
        }
        System.out.println(white+" "+black+" "+gray);
        // for(int i = 995; i<1010; i++){
        //    System.out.print(first[i]+" ");
        // } 
    }
}